pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentReplicator.java

```diff
@@ -176,7 +180,13 @@ private static final class ProducerSendCallback implements SendCallback {
         @Override
         public void sendComplete(Throwable exception, OpSendMsgStats opSendMsgStats) {
             if (exception != null) {
-                log.error("[{}] Error producing on remote broker", replicator.replicatorId, exception);
+                Throwable actEx = FutureUtil.unwrapCompletionException(exception);
+                if (actEx instanceof PulsarClientException.ProducerQueueIsFullError) {
+                    log.error("[{}] Discard to replicate non-persistent messages to the remote cluster because the"
```

Comment:
```suggestion
                    log.warn("[{}] Discard to replicate non-persistent messages to the remote cluster because the"
```

Reply:
warning should be fine since there is no at-least-once sematics for non-partitioned topic geo-replication.